use C.struct, not _Ctype_struct

Using cgo with Go 1.12 fails

```
# github.com/andelf/go-curl
../src/github.com/andelf/go-curl/easy.go:408:22: identifier "_Ctype_struct_curl_slist" may conflict with identifiers generated by cgo
```

references:
* https://github.com/google/gopacket/pull/589
* https://github.com/weaveworks/weave/issues/3577
* https://github.com/golang/go/issues/28721

--- a/easy.go
+++ b/easy.go
@@ -405,7 +405,7 @@ func (curl *CURL) Getinfo(info CurlInfo) (ret interface{}, err error) {
 		debugf("Getinfo %s", ret)
 		return ret, err
 	case C.CURLINFO_SLIST:
-		a_ptr_slist := new(_Ctype_struct_curl_slist)
+		a_ptr_slist := new(C.struct_curl_slist)
 		err := newCurlError(C.curl_easy_getinfo_slist(p, cInfo, &a_ptr_slist))
 		ret := []string{}
 		for a_ptr_slist != nil {

